### ISSUE TITLE
Suggestion: pass-through all static properties when wrapping SFC

### ISSUE THREAD

**Issue by geelen** [mobxjs/mobx-react#427]
Just ran into a bit of a bug, where I was trying to annotate one of my SFCs with some static properties:

```js
const Comp => () => (
  ...
)

Comp.someMetadata = ...

export default observer(Comp)
```

The static property `someMetadata` wasn't being copied across to the result of `observer`. I can fix the problem by hoisting `observer` up to the definition (e.g. `const Comp = observer(() => (...))`) but that's not going to be obvious to everyone, so I can see that being a source of bugs in the future.

I found the [relevant lines of code](https://github.com/mobxjs/mobx-react/blob/master/src/observer.js#L318-L321):

```js
return observer(
  class extends Component {
    static displayName = componentClass.displayName || componentClass.name
    static contextTypes = componentClass.contextTypes
    static propTypes = componentClass.propTypes
    static defaultProps = componentClass.defaultProps
    render() {
        return componentClass.call(this, this.props, this.context)
    }
  }
)
```

If this was changed to iterate over `Object.keys(componentClass)` and copy all static properties across, this problem would go away. And, given that React's context system is changing in the near future, maybe a more maintainable option for non-custom static properties too.

Happy to PR this if there's interest.

**Comment by mweststrate** [comment]
I think we already use hoist-statics for `inject`. Would the same mechanism work here? https://github.com/mobxjs/mobx-react/blob/409f3f2c274c113a0f1919daf81991816930b396/src/inject.js#L76

**Comment by geelen** [comment]
Yep, that seems like it'd do it!

**Comment by mweststrate** [comment]
Pr including test is welcome 😊

Op do 22 feb. 2018 23:55 schreef Glen Maddern <notifications@github.com>:

**Comment by mweststrate** [comment]
Fixed and released as mobx-react@5.1.0
